packages/gatsby-theme-store/src/components/ProductPage/SEO/useProductJsonLd.ts

```diff
@@ -0,0 +1,87 @@
+import { useLocation } from '@reach/router'
+import { graphql } from 'gatsby'
+import { useMemo } from 'react'
+import type { ComponentPropsWithoutRef } from 'react'
+import type { ProductJsonLd } from 'gatsby-plugin-next-seo'
+
+import { useCurrency } from '../../../sdk/localization/useCurrency'
+import type { ProductPageProps } from '../../../templates/product'
+
+type Options = ProductPageProps
+
+type Return = ComponentPropsWithoutRef<typeof ProductJsonLd> | null
+
+export const useProductJsonLd = (options: Options): Return => {
+  const {
+    data: {
+      vtex: { product },
+    },
+  } = options
+
+  const { host, pathname } = useLocation()
+  const [currency] = useCurrency()
+
+  return useMemo(() => {
+    if (product == null) {
+      return null
+    }
+
+    const { productName: name, items, description, brand } = product
+
+    const [sku] = items!
+    const images = sku!.images!.map((i) => i!.imageUrl!)
+    const seller = sku!.sellers?.[0]
+
+    if (!sku || !images || !seller || !brand) {
+      return null
+    }
```

Comment:
This doesn't make sense, or remove this `!` operator or remove `!sku` from the if statement.

You assume that's defined and check if doesn't exist. Maybe you should change the order of this code

```suggestion
    if (!sku || !brand) {
      return null
    }
    
        const images = sku!.images!.map((i) => i!.imageUrl!)
    const seller = sku!.sellers?.[0]
    
    if (!images || !seller) { return null }
```

Reply:
I'm also not a fan of `!` assertions, kind of defeats the purpose of TS and I've seen errors caused by it more than once, on SFJ even.

Even if you are sure it's going to be defined, I'd use `?.` instead and have it return if nil

Reply:
nice @igorbrasileiro , this makes sense. However, I don't like using `?.` operator since this adds unnecessary branches. I'll just remove the `!sku` since it doesn't need to exist